workflow/controller/dag.go

```diff
@@ -377,12 +377,14 @@ func (woc *wfOperationCtx) executeDAGTask(dagCtx *dagContext, taskName string) {
 
 	node := dagCtx.getTaskNode(taskName)
 	task := dagCtx.GetTask(taskName)
-	if node != nil && node.Completed() {
-		// Run the node's onExit node, if any.
-		hasOnExitNode, onExitNode, err := woc.runOnExitNode(task.Name, task.OnExit, dagCtx.boundaryID, dagCtx.tmplCtx)
-		if hasOnExitNode && (onExitNode == nil || !onExitNode.Completed() || err != nil) {
-			// The onExit node is either not complete or has errored out, return.
-			return
+	if node != nil && node.Fulfilled() {
+		if node.Completed() {
```

Comment:
can you double-check this logic please? this has changed - which I think you intended - but please double check again

Reply:
This is intended, thanks for asking to make sure.